From ced29adf705bae46f440387a735fd7aaa06526a3 Mon Sep 17 00:00:00 2001
From: Catalin Enache <catalin.enache@windriver.com>
Date: Fri, 9 Jun 2017 10:51:33 +0300
Subject: [PATCH] lxc-sshd: mount /etc/init.d only if it exists

This will avoid mounting errors such as:

lxc-start 20170531032501.763 ERROR    lxc_utils -
 ../../../lxc-2.0.4/src/lxc/utils.c:safe_mount:1692 -
No such file or directory - Failed to mount /etc/init.d
onto /usr/lib64/lxc/rootfs/etc/init.

Upstream-Status: Pending

Signed-off-by: Catalin Enache <catalin.enache@windriver.com>
---
 templates/lxc-sshd.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/templates/lxc-sshd.in b/templates/lxc-sshd.in
index 9b07ea6..94585d3 100644
--- a/templates/lxc-sshd.in
+++ b/templates/lxc-sshd.in
@@ -143,11 +143,16 @@ lxc.mount.entry = /usr usr none ro,bind 0 0
 lxc.mount.entry = /sbin sbin none ro,bind 0 0
 lxc.mount.entry = tmpfs var/run/sshd tmpfs mode=0644 0 0
 lxc.mount.entry = @LXCTEMPLATEDIR@/lxc-sshd $init_path none ro,bind 0 0
-lxc.mount.entry = /etc/init.d etc/init.d none ro,bind 0 0
 
 lxc.mount.auto = cgroup:mixed proc:mixed sys:mixed
 EOF
 
+    if [ -d /etc/init.d ]; then
+        cat <<EOF >> $path/config
+lxc.mount.entry = /etc/init.d etc/init.d none ro,bind 0 0
+EOF
+    fi
+
     # Oracle Linux and Fedora need the following two bind mounted
     if [ -d /etc/sysconfig/network-scripts ]; then
         cat <<EOF >> $path/config
-- 
